### ISSUE TITLE
Mapbox crashes when using "aboveLayerID" property

### ISSUE THREAD

**Issue by systemlevel** [react-native-mapbox/maps#52]
**Describe the bug**
Mapbox crashes when using "aboveLayerID" property.

**To Reproduce**

Add various layers and set one of the layers to appear "above" the other as follows:

```
        <Mapbox.VectorSource
          id="customSourceExample"
          url={VECTOR_SOURCE_URL}
        >
          <Mapbox.FillLayer
            id="customSourceFill"
            sourceLayerID="roadless"
            style={mbStyles.boxFill}
            aboveLayerID="basemap"
          />
        </Mapbox.VectorSource>


       <Mapbox.RasterSource id="rasterSourceBasemap" tileSize={256} url={url}>
          <Mapbox.RasterLayer
            id="basemap"
            sourceID="rasterSourceBasemap"
            style={{rasterOpacity: 1}}
          />
        </Mapbox.RasterSource>
```

**Expected behavior**
In the above example, the Vector FillLayer should appear above the RasterLayer based on the property defined aboveLayerID="basemap"

**Screenshots**
xCode crashes:

<img width="971" alt="Screen Shot 2019-05-18 at 10 56 49 PM" src="https://user-images.githubusercontent.com/894983/57977966-40d6a000-79c0-11e9-944e-7decca6b102a.png">

**Versions (please complete the following information):**
 - Platfrom: iOS
 - Device: XR Simulator
 - React Native Version 0.59

**Comment by mfazekas** [comment]
See also https://github.com/nitaliano/react-native-mapbox-gl/issues/1135.

The issue is that at the time `#customSourceFill` added to the map, the #aboveLayerID is not yet added to the map therefore it will not find the layer and that's the cause of crash.

Ideally in such cases we'd either defer the insertion until the referred layer is added

**Comment by systemlevel** [comment]
@mfazekas Thank you for that insight.  What is the best way to ensure a layer is added to the map? Does the order in which I define the layers have any impact on that order?

For example, if I were to switch these layers around:

```
       <Mapbox.RasterSource id="rasterSourceBasemap" tileSize={256} url={url}>
          <Mapbox.RasterLayer
            id="basemap"
            sourceID="rasterSourceBasemap"
            style={{rasterOpacity: 1}}
          />
        </Mapbox.RasterSource>

        <Mapbox.VectorSource
          id="customSourceExample"
          url={VECTOR_SOURCE_URL}
        >
          <Mapbox.FillLayer
            id="customSourceFill"
            sourceLayerID="roadless"
            style={mbStyles.boxFill}
            aboveLayerID="basemap"
          />
        </Mapbox.VectorSource>
```

Would that load in the proper order?

**Comment by mfazekas** [comment]
@systemlevel yep that should work.

**Comment by mfazekas** [comment]
@systemlevel with #57 you should be able to declare in any order and RNMapbox should wait for the layer to be inserted bellow.

**Comment by systemlevel** [comment]
@mfazekas Excellent update. Thank you. That is perfect!

**Comment by systemlevel** [comment]
@mfazekas Doing some more testing on this tonight.  I'm still having some issues getting the aboveLayerID property working.  

Here is some more context:

1) We have made a custom style map in mapbox studio
2) We have added various vector layers to that mapbox studio map (these are layers we uploaded into studio)
3) When I try to order those layers in the react-native-mapbox-gl client they don't seem to respond to the order I try to define whether it be via aboveLayerID or layerIndex.

I'm able to control the visibility but I'm unable to control where they sit in relation to the RasterSource layer:

```
       <Mapbox.RasterSource id="rasterSourceBasemap" tileSize={256} url={url}>
          <Mapbox.RasterLayer
            id="basemap"
            sourceID="rasterSourceBasemap"
            style={{rasterOpacity: 1}}
          />
        </Mapbox.RasterSource>
```
